### ISSUE TITLE
Get input blur event handler

### ISSUE THREAD

**Issue by amardeo1** [xtrinch/react-touch-screen-keyboard#39]
Is it possible to have a call back for input on blur event just like on change event. I do not see anything regarding onBlur call back on the documentation. I am not able to figure out how to do it.

Is it possible to do it? If not can you please add it.

Thanks in advance.

**Comment by reganlawton** [comment]
@amardeo1 I'll look into getting a callback in the next update

**Comment by amardeo1** [comment]
@reganlawton Thank you for looking into it. Do you have any timeline when it will be done. We desperately need it in my project. Because of this we are not able to do any validations on focus out.

Also if possible it would be great to have a callback for for focus as well. Also it would be great to be able to add attributes like required to input.

Thank you so much in advance.

**Comment by amardeo1** [comment]
@reganlawton Thank you for looking into it. Do you have any timeline when it will be done. We desperately need it in my project. Because of this we are not able to do any validations on focus out.

Also if possible it would be great to have a callback for for focus as well. Also

**Comment by reganlawton** [comment]
@amardeo1 I have a PR if you can take a look and see if that will meet your requirements.

**Comment by amardeo1** [comment]
@reganlawton Thank you Regan.  It looks good and you have incorporated what I requested. Please merge the PR.
I have one more request. If it is possible to do. Since the keyboard appears right below the input, user has to click outside then click on input (as keyboard is on top of other bottom input elements), it gives bad user experience (when I add position absolute to the keyboard). When I add position relative to the keyboard it shifts the whole page down (this way user do not have to click outside, as input elements are shifted down). That is not desirable either. I also tried to display the keyboard at the bottom, always in the same place (I applied position fixed and bottom 0). But with this approach, it always shows at the bottom, but it moves left and right.

So, is there an option to attach the keyboard to the body itself, instead of attaching in parallel to the input. This can be a configurable option.

If this can be done, it will resolve all our issues.

Please let me know if it is possible to do.


Thanks,
Amar

**Comment by reganlawton** [comment]
@amardeo1 did you want to open a new issue for that request? I thinking we might be able to pass through a keyboard component, but will need more planning.
